### PR TITLE
feat: implement v2 list subscription api keys

### PR DESCRIPTION
**Pull request by jhaeyaert**
## Issue

https://gravitee.atlassian.net/browse/APIM-1234

## Description

This PR implement the list subscription api keys.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-eegwavdgls.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -3111,91 +3134,96 @@ components:
...
+                        example: 2023-05-20T12:30:42.101Z
+                    endingAt:
+                        type: string
+                        format: date-time
+                        description: The datetime when the subscription ends. No ending date means the subscription never ends.
+                        example: 2023-05-25T12:40:46.184Z
+                    commentMessage:
+                        type: string
+                    commentRequired:
+                        type: boolean
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml]: "Comment message" is a message from the API Publisher displayed to the consumer at subscription time.
Example: Explain why you want to subscribe

"Comment required" is a boolean to determine if the consumer has to write a "consumer message" or if he/she can leave the field blank.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/fixtures/ApplicationFixtures.java [hunk 2]
```diff
@@ -30,6 +30,18 @@
  */
 public class ApplicationFixtures {
 
+    private static final PrimaryOwnerEntity.PrimaryOwnerEntityBuilder BASE_PRIMARY_OWNER_ENTITY = PrimaryOwnerEntity
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/fixtures/ApplicationFixtures.java]: We already have a `PrimaryOwnerFixtures` class. I think you could use it.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/SubscriptionMapper.java [hunk 3]
```diff
@@ -75,4 +75,8 @@ public interface SubscriptionMapper {
     @Mapping(target = "id", expression = "java(subscriptionId)")
     @Mapping(target = "plan", source = "transferSubscription.planId")
     TransferSubscriptionEntity map(TransferSubscription transferSubscription, String subscriptionId);
+
+    ApiKey map(ApiKeyEntity apiKeyEntity);
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/SubscriptionMapper.java]:
```suggestion
    ApiKey mapApiKey(ApiKeyEntity apiKeyEntity);
```

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml (hunk 1 above)
Comment (jhaeyaert) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml]: It is a mistake. Those attributes belong to the `plan` entity. I've updated the openapi spec to reflect the changes on them.

**Review by phiz71 — APPROVED** [review]

**Pull request merged**